images.CI/macos/select-datastore.ps1

```diff
@@ -54,14 +54,14 @@ function Select-DataStore {
 
     # 1. Name starts with ds-local-Datastore
     # 2. FreespaceGB > 400 Gb
-    # 3. VM count on a datastore < 2
+    # 3. Choose a datastore with the minimal VM count < 2
 
     Write-Host "Start Datastore selection process..."
     $allDatastores = Get-Datastore -Name $templateDatastore | Where-Object { $_.State -eq "Available" }
     $buildDatastore = $allDatastores | Where-Object { $_.FreeSpaceGB -ge $thresholdInGb } | Where-Object {
         $vmOnDatastore = @((Get-ChildItem -Path $_.DatastoreBrowserPath).Name -notmatch "^\.").Count
         $vmOnDatastore -lt $vmCount
-    } | Get-Random | Select-Object -ExpandProperty Name
+    } | Group-Object -Property { $vmOnDatastore } | Select-Object -First 1 -ExpandProperty Group | Get-Random |  Select-Object -ExpandProperty Name
```

Comment:
I am not sure that `$vmOnDatastore` will be available here.
This variable was defined in `Where-Object` and it won't be saved / passed to next pipeline.

Reply:
Sorry, I was wrong.
It will be available. But It looks a bit strange.

Checked on simple example:
```
$arr = @(
    @{ Name = "aaa" },
    @{ Name = "bbb" },
    @{ Name = "ccc" }
)

$arr | Where-Object {
    $myVariable = $_.Name -eq "bbb"
    return $true
} | ForEach-Object {
    Write-Host "Hello $myVariable"
}
```